flocker/control/httpapi.py

```diff
@@ -812,11 +818,15 @@ def replace_configuration(self, applications, deployment):
         :param deployment: Configuration of which applications run on
             which nodes.
         """
+        # XXX we're going to have to deal with UUIDs better,
+        # somehow... look up state of cluster? Change deployment
+        # configuration format?
         try:
             configuration = FigConfiguration(applications)
             if not configuration.is_valid_format():
                 configuration = FlockerConfiguration(applications)
             return self.persistence_service.save(model_from_configuration(
+                self.cluster_state_service.as_deployment(),
```

Comment:
The cluster state now gets passed to `model_from_configuration` so that supplied IP addresses can be matched to node_uuids.
What if agents have not yet reported their state?